### PR TITLE
MGMT-7351 - Use test-infra deploy scripts that were transferred from assisted-service

### PR DESCRIPTION
**Pull request by eliorerz**

### CODE REVIEW

File: ci-operator/step-registry/baremetalds/assisted/operator/gather/baremetalds-assisted-operator-gather-ref.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ref:
   as: baremetalds-assisted-operator-gather
-  from: assisted-service-scripts
```
Comment (osherdp) [inline ci-operator/step-registry/baremetalds/assisted/operator/gather/baremetalds-assisted-operator-gather-ref.yaml:3]: can you remove generation of ``assisted-service-scripts`` image on the branch configuration?

File: ci-operator/step-registry/baremetalds/assisted/operator/gather/baremetalds-assisted-operator-gather-commands.sh [hunk 2]
```diff
@@ -15,8 +15,6 @@ fi
 # shellcheck source=/dev/null
 source "${SHARED_DIR}/packet-conf.sh"
 
-tar -czf - . | ssh "${SSHOPTS[@]}" "root@${IP}" "cat > /root/assisted-service.tar.gz"
```
Comment (osherdp) [inline ci-operator/step-registry/baremetalds/assisted/operator/gather/baremetalds-assisted-operator-gather-commands.sh:18]: each step should be independent from the previous ones, at least on the technical level
so you'll have to handle that some way or another...

File: ci-operator/step-registry/baremetalds/assisted/operator/setup/baremetalds-assisted-operator-setup-commands.sh [hunk 3]
```diff
@@ -10,7 +10,7 @@ echo "************ baremetalds assisted operator setup command ************"
 # shellcheck source=/dev/null
 source "${SHARED_DIR}/packet-conf.sh"
 
-tar -czf - . | ssh "${SSHOPTS[@]}" "root@${IP}" "cat > /root/assisted-service.tar.gz"
+#tar -czf - . | ssh "${SSHOPTS[@]}" "root@${IP}" "cat > /root/assisted-service.tar.gz"
```
Comment (osherdp) [inline ci-operator/step-registry/baremetalds/assisted/operator/setup/baremetalds-assisted-operator-setup-commands.sh:13]: ?

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by osherdp** [comment]
/cc @pawanpinjarkar 
wanted you to considerate those changes in case there's some work still being done around upgrade jobs

File: ci-operator/step-registry/baremetalds/assisted/operator/setup/baremetalds-assisted-operator-setup-commands.sh (hunk 3 above)
Comment (eliorerz) [reply ci-operator/step-registry/baremetalds/assisted/operator/setup/baremetalds-assisted-operator-setup-commands.sh:13]: Need to delete it

File: ci-operator/step-registry/baremetalds/assisted/operator/gather/baremetalds-assisted-operator-gather-ref.yaml (hunk 1 above)
Comment (lranjbar) [reply ci-operator/step-registry/baremetalds/assisted/operator/gather/baremetalds-assisted-operator-gather-ref.yaml:3]: There is another dependency to change for `assisted-service-scripts` -- can we hold on this part?

File: ci-operator/step-registry/baremetalds/assisted/operator/gather/baremetalds-assisted-operator-gather-commands.sh (hunk 2 above)
Comment (lranjbar) [reply ci-operator/step-registry/baremetalds/assisted/operator/gather/baremetalds-assisted-operator-gather-commands.sh:18]: 🤔 This is copying the files from assisted-service repo to the remote server. Instead it should be the files from assisted-test-infra right?

File: ci-operator/step-registry/baremetalds/assisted/operator/gather/baremetalds-assisted-operator-gather-ref.yaml (hunk 1 above)
Comment (osherdp) [reply ci-operator/step-registry/baremetalds/assisted/operator/gather/baremetalds-assisted-operator-gather-ref.yaml:3]: sure thing
I wasn't aware of publish step using it, so thought we don't need that anymore

**Comment by lranjbar** [comment]
Okay looking at these failures I believe this is breaking down on the following:

1) When we start the step we are running in a pod on the CI cluster
2) Most of these actions in the test need to happen on the remote baremetal server. This is why we place the repo files in a tar and SSH them over.
3) In the past this repo file copy was assisted-service but now it should be assisted-test-infra.

The fix for this current failure is we still need this just change to assisted-test-infra instead of removing it.

Moving forward we do this tar and SSH copy in every workflow we have on the baremetal servers... I think we can isolate this copying of files from the repo to the remote server to its own step at some point. That would make our steps more reusable. LMK if this makes sense.

**Comment by eliorerz** [comment]
Currently it is not prioritized

**Pull request closed without merging**